cdap-ui/app/cdap/components/PipelineTriggers/ScheduleRuntimeArgs/index.js

```diff
@@ -40,19 +44,56 @@ export default class ScheduleRuntimeArgs extends Component {
   };
 
   componentDidMount() {
-    let {id: triggeringPipelineId, namespace: triggeringPipelineNS} = this.props.triggeringPipelineInfo;
-    let {id: triggeredPipelineId, namespace: triggeredPipelineNS} = this.props.triggeredPipelineInfo;
-    fetchPipelineMacroDetails(triggeringPipelineId, triggeringPipelineNS);
-    fetchPipelineMacroDetails(triggeredPipelineId, triggeredPipelineNS, true);
-
-    ScheduleRuntimeArgsStore.subscribe(() => {
+    this.sub = ScheduleRuntimeArgsStore.subscribe(() => {
       let {triggeringPipelineInfo, argsMapping} = ScheduleRuntimeArgsStore.getState().args;
       let {macros, configStages} = triggeringPipelineInfo;
+
       this.setState({macros, configStages, argsMapping});
     });
+
+    if (this.props.disabled) {
+      ScheduleRuntimeArgsStore.dispatch({
+        type: SCHEDULERUNTIMEARGSACTIONS.SETDISABLED
+      });
+
+      let {scheduleInfo} = this.props;
+
+      let triggerProperties = objectQuery(scheduleInfo, 'properties', 'triggering.properties.mapping');
+
+      try {
+        triggerProperties = JSON.parse(triggerProperties);
+      } catch (e) {
+        console.log('properties are not JSON');
+        return;
+      }
+
+      let runtimeArgs = triggerProperties.arguments;
+      if (runtimeArgs.length > 0) {
+        runtimeArgs.forEach((args) => {
+          setArgMapping(args.source, args.target, 'runtime');
```

Comment:
Might not for small mapping but for mappings around 10 - 15 this might repeatedly call subscribers for each iteration in the loop which will cause line 51 to call setState. Might be good to see if adding a batch add action to the store helps.